src/CoreShop/Bundle/OrderBundle/Pimcore/Repository/CartRepository.php

```diff
@@ -41,6 +41,7 @@ public function findNamedForCustomer(CustomerInterface $customer, $name): ?Order
     {
         $list = $this->getList();
         $list->setCondition('customer__id = ? AND name = ? AND order__id is null', [$customer->getId(), $name]);
+        $list->setLimit(1);
```

Comment:
is not right here, named can return multiple carts

Reply:
hm, but then is the following code below wrong:
`if (count($objects) === 1 && $objects[0] instanceof OrderInterface) {
            return $objects[0];
        }`
because only one cart will returned?

Reply:
is this function somewhere in use @dpfaffenbauer? i can't find a usage in coreshop, if not then how about to return the found collection of carts?

Reply:
true, it always ever return if it is one. We don't have any usage of it CoreShop. I don't think it even works... So we should remove it with 3.0